heap/max_heap.py

```diff
@@ -71,30 +79,32 @@ def _bubble_up(self, index):
         # * `_sift_down` grabs the indices of this element's children and determines which child has a larger value. 
        If the larger child's value is larger than the parent's value, the child element is swapped with the parent."""
     def _sift_down(self, index):
+        left_child = (index * 2) + 1
+        right_child = (index * 2) + 2
+        parent = (index - 1) // 2
         # if LC greater than length of array or RC greater than length of array
-        pass
-            
+        if left_child > len(self.storage) or right_child > len(self.storage):
+            return
+
+        if self.storage[left_child] > self.storage[right_child]:
+            self.storage[left_child], self.storage[parent] = self.storage[parent], self.storage[left_child]
+            return self._sift_down(index)
```

Comment:
you're getting stuck in an infinite loop becaues `index` never changes :) you're passing in the same value of `index` every time, so you always get the same values for `left_child`, `right_child`, and `parent`